### PR TITLE
Add auto_select2 and auto_awesomplete ruby gem

### PR DESCRIPTION
**Pull request by skorth**
I thought about creating an Issue instead of a PR cause i am not 100% sure if this security fix is worth it. There is no CVE/OSVDB, so i just played around with the numbering system https://github.com/rubysec/ruby-advisory-db/issues/224, too.

### CODE REVIEW

**Comment by reedloden** [comment]
Requested a [CVE](http://www.openwall.com/lists/oss-security/2016/01/11/2) and OSVDB assignment. Will update once I hear something back.

**Comment by reedloden** [comment]
Note that this also affects `auto_awesomplete`, so need to add an advisory for it as well.

File: gems/auto_select2/RUBYSEC-2016-0001.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+gem: auto_select2
+rubysec: 2016-0001
+url: https://github.com/Loriowar/auto_select2/issues/4
+title: |
+  Search functionality allows arbitrary search execution
+date: 2016-1-8
+description: |
+  auto_select2 Gem for Ruby contains a flaw that is triggered when handling the
+  'params[:default_class_name]' option. This allows users to search any object
+  of all given ActiveRecord classes
```
Comment (JuanitoFatas) [inline gems/auto_select2/RUBYSEC-2016-0001.yml]: Missing period :wink: ?

File: gems/auto_select2/RUBYSEC-2016-0001.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+gem: auto_select2
+rubysec: 2016-0001
+url: https://github.com/Loriowar/auto_select2/issues/4
+title: |
+  Search functionality allows arbitrary search execution
+date: 2016-1-8
```
Comment (reedloden) [inline gems/auto_select2/RUBYSEC-2016-0001.yml]: 2016-01-08

File: gems/auto_select2/RUBYSEC-2016-0001.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+gem: auto_select2
+rubysec: 2016-0001
+url: https://github.com/Loriowar/auto_select2/issues/4
+title: |
+  Search functionality allows arbitrary search execution
```
Comment (reedloden) [inline gems/auto_select2/RUBYSEC-2016-0001.yml]: I personally include the gem name in the title, but not required... We should probably standardize.

Comment (VanessaHenderson) [inline gems/auto_select2/RUBYSEC-2016-0001.yml]: Standardizing titles is incredibly hard.

Comment (phillmv) [inline gems/auto_select2/RUBYSEC-2016-0001.yml]: Hrm yeah okay I'd write a style guide

**Comment by skorth** [comment]
Thx for the Feedback! I will update as soon as possible.

**Comment by reedloden** [comment]
OSVDB guys are generally pretty quick, so I should have an identifier for you to use by EOD Monday, I bet.

**Comment by reedloden** [comment]
Use OSVDB-132800 for both of these.

**Comment by reedloden** [comment]
@skorth poke on this? Happy to take this over if you're too busy.

**Comment by skorth** [comment]
@reedloden done, please double check. How did you get the OSVDB id?

File: gems/auto_select2/OSVDB-132800.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+---
+gem: auto_select2
+osvdb: 132800
+url: http://www.openwall.com/lists/oss-security/2016/01/11/2
```
Comment (reedloden) [inline gems/auto_select2/OSVDB-132800.yml]: https://github.com/Loriowar/auto_select2/issues/4

File: gems/auto_awesomplete/OSVDB-132800.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+---
+gem: auto_awesomplete
+osvdb: 132800
+url: http://www.openwall.com/lists/oss-security/2016/01/11/2
```
Comment (reedloden) [inline gems/auto_awesomplete/OSVDB-132800.yml]: Use https://github.com/Tab10id/auto_awesomplete/issues/2 here

**Comment by reedloden** [comment]
@skorth just swap the URLs, and should be good to go. I prefer to keep stuff at the vendor level. My post to oss-security@ just summed up the issue, but it wasn't made or done by the vendor.

As for the OSVDB ID, I e-mailed moderators [at] osvdb.org, and they assigned one.

**Comment by skorth** [comment]
@reedloden Your right, there was a discussion about the url. Always vendor if possible. Thx!

**Comment by skorth** [comment]
@reedloden done

**Comment by reedloden** [comment]
Awesome! Thanks, @skorth

**Pull request merged**